test/integration/acceptance/create_account_test.cpp

```diff
@@ -156,3 +160,35 @@ TEST_F(CreateAccount, EmptyName) {
                   empty_name, kDomain, kNewUserKeypair.publicKey())),
               CHECK_STATELESS_INVALID);
 }
+
+/**
+ * Checks that there is no privelege elevation issue via CreateAccount
+ *
+ * @given two domains: the first domain has default role that contain
+ * can_create_account permission, the second domain has default role that
+ * contains more permissions than default role of the first domain
+ * @when the user of an account from the first domain tries to create an account
+ * in the second domain
+ * @then transaction should fail stateful validation
+ */
+TEST_F(CreateAccount, PrivelegeElevation) {
+  auto second_domain_tx = complete(
+      baseTx(kAdminId).createDomain(kSecondDomain, kAdminRole), kAdminKeypair);
+  auto create_elevated_user = complete(baseTx().createAccount(
+      kNewUser, kSecondDomain, kNewUserKeypair.publicKey()));
+  auto rejected_hash = create_elevated_user.hash();
+
+  IntegrationTestFramework(1)
+      .setInitialState(kAdminKeypair)
+      .sendTxAwait(second_domain_tx, check(1))
+      .sendTxAwait(makeUserWithPerms(), check(1))
+      .sendTx(create_elevated_user)
+      .skipProposal()
+      .checkVerifiedProposal(check(0))
+      .checkBlock([&rejected_hash](const auto &block) {
+        const auto hashes = block->rejected_transactions_hashes();
```

Comment:
The code in this lambda can be simplified using gmock's means: 

```suggestion
         ASSERT_THAT(block->rejected_transactions_hashes(), Contains(rejected_hash));
```